src/main/java/hudson/plugins/active_directory/ActiveDirectoryDomain.java

```diff
@@ -207,7 +207,7 @@ public Attribute getRecordFromDomain(){
         Attribute a = null;
         try {
             LOGGER.log(Level.FINE, "Attempting to resolve {0} to NS record", name);
-            ictx = createDNSLookupContext();
+            ictx = DNSUtils.createDNSLookupContext();
```

Comment:
code was duplicated in 2 places.  extracted it to a helper and made the helper more amenable to testing.